### ISSUE TITLE
Provide support for direct mapping to JSON string.

### ISSUE THREAD

**Issue by B-Esmaili** [dotnet/SqlClient#424]
In some languages (like PHP for example)  there is possibility to directly map query result into `JSON` string which makes a lot of  sense when we don't need to manipulate the result returned from query execution which would result in better performance due to eliminating extra step of mapping to application models (and then mapping to JSON string later on).Is there any plan to add such a feature to `SqlClient`?

**Comment by cheenamalhotra** [comment]
Hi @B-Esmaili 

We've never received this request before so as of now there are no plans to introduce this feature. 
Also the current conversion strategy for conversion is SqlDataReader >> DataTable >> JSON, and takes additional step, as you also mentioned.

Maybe something to consider in future, unless there's a heavy demand for it.
Feel free to contribute a PR if you'd like to!

**Comment by B-Esmaili** [comment]
@cheenamalhotra thanks for your quick response , but i thinks we are already in `JSON` era and in evaluating features of a language `JSON` support is one of many important factors. Nevertheless I hope more developers would endorse mentioned requirement and move it to higher priority.

**Comment by Wraith2** [comment]
This is a job normally done by several kinds of mappers and wrappers like EFCore or Drapper. While the same sort of functionality could be added into SqlClient there's nothing it can do that those projects (and others) don't already do. Why would it be a good idea to add the functionality at this level?

**Comment by B-Esmaili** [comment]
@Wraith2 could you point me to a link containing sample code or doc for either Dapper or EFCore?Although i am not interested in EFCore but i am really curious about thing i've missed for so many years!

**Comment by Wraith2** [comment]
Looking at the scenario you described you probably want something like this https://github.com/AutoMapper/AutoMapper.Data which isn't an orm just a mapper.

**Comment by B-Esmaili** [comment]
I couldn't find anything related to mapping to JSON  in the provided link.

**Comment by Wraith2** [comment]
No, that would be the job of a serialization library. Can you give a code example of what you're trying to do? because it sounds bonkers to me.

**Comment by Georgy-Losenkov** [comment]
Why don't you want to perform conversion to JSON using T-SQL?
FOR JSON clause is available starting SQL SERVER 2016
[https://docs.microsoft.com/en-us/sql/relational-databases/json/format-query-results-as-json-with-for-json-sql-server?view=sql-server-ver15](url)

**Comment by Wraith2** [comment]
I'd still like an example use case or two to clarify exactly what functionality is desiredhere.

Json support in sql server returns a string and then to do anything with that string you're going to have to parse it. So you send a query to the engine which does a query then all the work of composing the results it into a string so you can receive it and then parse the json apart. I'd cut out the string format in the middle if possible and work with the data. It just seems very wasteful to me.

If you're going to do things like sending query results straight to or from apps into the sql server I don't think allowing you to bypass the word of validating the data is a good idea.

**Comment by Georgy-Losenkov** [comment]
@Wraith2, I completely agree with you. I am sorry that I have not addressed question to correct person.

**Comment by cheenamalhotra** [comment]
Closing as the feature has not received enough interest from community all this time.
